hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/multipart/S3InitiateMultipartUploadRequestWithFSO.java

```diff
@@ -118,10 +118,13 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       // check if the directory already existed in OM
       checkDirectoryResult(keyName, pathInfoFSO.getDirectoryResult());
 
+      final OmBucketInfo bucketInfo = getBucketInfo(omMetadataManager,
+          volumeName, bucketName);
+
       // add all missing parents to dir table
       missingParentInfos = OMDirectoryCreateRequestWithFSO
-          .getAllMissingParentDirInfo(ozoneManager, keyArgs, pathInfoFSO,
-              transactionLogIndex);
+          .getAllMissingParentDirInfo(ozoneManager, keyArgs, bucketInfo,
+              pathInfoFSO, transactionLogIndex);
```

Comment:
Down while create omKeyInfo, setAcl is not done using below, its done using 'OzoneAclUtil.fromProtobuf(keyArgs.getAclsList())'

```
getAclsForKey(keyArgs, bucketInfo, pathInfo,
              ozoneManager.getPrefixManager())
```